test/functional/p2p_add_connections.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test add_outbound_p2p_connection test framework functionality"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.p2p import P2PInterface
+from test_framework.util import assert_equal
+
+
+class P2PAddConnections(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        self.disconnect_nodes(0, 1)
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+
+        self.log.info("Add 2 block-relay-only connections to node 1")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[1].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
```

Comment:
50720f411e05cf1bba fun nit, can use `product` here (thanks to @jnewbery for the tip a few months ago) to replace these two blocks:
```diff
+from itertools import product
...
-        self.log.info("Add 2 block-relay-only connections to node 0")
-        for i in range(2):
-            self.log.info("block-relay-only: {}".format(i))
-            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
-
-        self.log.info("Add 2 block-relay-only connections to node 1")
-        for i in range(2):
-            self.log.info("block-relay-only: {}".format(i))
-            self.nodes[1].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+        self.log.info("Add 2 block-relay-only connections to nodes 0 and 1")
+        for node, conn in product(range(2), range(2)):
+            self.log.info("node {}, block-relay-only conn {}".format(node, conn))
+            self.nodes[node].add_outbound_p2p_connection(P2PInterface(), connection_type="BlockRelay")
```

Reply:
("BlockRelay" in the suggestion because I was testing the case-insensitive code above)

Reply:
thanks but I am a pass, within the context of this test, I find this a bit less clear in terms of test structure (putting together node 0 & 1) and logging.